WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentUseCase.swift

```diff
@@ -228,7 +228,15 @@ private extension CollectOrderPaymentUseCase {
         order.datePaid == nil
     }
 
-    func checkOrderIsStillEligibleForPayment(onCheckCompletion: @escaping (Result<Void, Error>) -> Void) {
+    func checkOrderIsStillEligibleForPayment(alertProvider paymentAlerts: CardReaderTransactionAlertsProviding,
```

Comment:
We use datePaid elsewhere for this and I agree that it should be improved, but I don't think that this is the right place to address it. For one thing, the main place it matters is on Order Details, where they won't even get a Collect Payment button if it's not nil.

This does feel bad; it's an avenue for confusion over whether an order is paid or not, because changing an order's status to `On Hold` then back to `Pending Payment` leaves it still showing as paid in full on order details, with the exception of the status.

The app doesn't change `datePaid`, from what I can tell, so this is something that needs to be addressed with both Woo Payments and perhaps core platform as well. Certainly we need to do the same thing on Android if we change it.